src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -322,6 +325,9 @@ private void HandleChangesForStaticIndexes(DatabaseRecord record, long index, Li
 
                 try
                 {
+                    if (definition.Staggered && ShouldSkipThisNode(record, name))
```

Comment:
Yes ok. I actually separated the check from the ShouldSkip function so that it will be easy to understand that it only runs on staggered. But I can also change the name of the function to ShouldSkipThisNodeWhenStaggered.